### ISSUE TITLE
Leaving Preview mode and coming back resets My Progress bar step, question count

### ISSUE THREAD

**Issue by JessicaFrank** [CCALI/a2jauthor#166]
Testing an issue for an author where they reported a problem with Step 1, Q8. In trying to figure out which actual question they were referring to, I went back and forth from authoring to preview mode several times. I was on a Step 1, Q8 when I left preview mode. When I came back after emailing them (but not leaving a2jauthor.org or clearing the debug panel or leaving that specific interview -- all of which I'd expect to clear the My Progress log), the My Progress log was cleared and reset to Step 1, Q1. 

I was on the same page I had been on when I clicked Edit This in preview mode, so the actual page doesn't change, just the history of my progress through the interview. 

The log still shows the pages I went to. All that's different is My Progress question numbering is not what it was/should be and my history is gone from the drop down My Progress as well.

Steps to replicate:

1. Run an interview in preview mode for a couple pages. Note My Progress step and question #. 
2. Go back into authoring with Edit this button.
3. Go back into preview mode from that question's Preview button. 
4. Note difference in My Progress. Should now say Step whatever, Q1.

**Comment by JessicaFrank** [comment]
This is in production Author 7.0.19-2021-02-16

**Comment by mitch-mitchel** [comment]
Just confirmed this against legacy CAJA code as I remembered it always had this behavior aka the myProgress bar reseting to to the new 'first' page visited. I agree that it should restore the visitedPages list/myprogress between Previews. 

@JessicaFrank  This should be fixed via the Advanced Nav changes, but wonder if it's important enough to fix/hotfix sooner rather than later? 

Otherwise, my other question is what other things between one preview and the next should be maintained that we might be currently missing/resetting, and among the list of things to persist, when should they reset.  For example, am I missing anything below:

Preserve
- MyProgress (really the source visitedPages list) - currently not working 
- Variables/Answers - works currently
- TraceMessages - works currently
- The page open in the QDE - preserved for Resume Edit, but cleared or reset when Edit This button or Preview Tab clicked (not in the QDE)


Reset
- MyProgress should reset on Interviews Tab visit like the Answers and Messages currently do?
- Vars/Answers and TraceMessages have explicit clear buttons as well - should MyProgress have a manual clear mechanic?

**Comment by mitch-mitchel** [comment]
did a quick exploration of this, and it's fairly easy to restore the list of pages, but it breaks how answers are bound to those navigated pages.  still doable, but not an easy fix so likely best handled at the same time the Advanced Nav changes come through

**Comment by JessicaFrank** [comment]
@mikemitchel I don't think this requires a hotfix. It only affects authors when testing in Preview, so it's not too big of a bug.
